src/i2c.rs

```diff
@@ -0,0 +1,217 @@
+use std::marker::PhantomData;
+
+use crate::bindings::{
+    CgosI2CCount, CgosI2CGetFrequency, CgosI2CGetMaxFrequency, CgosI2CIsAvailable, CgosI2CRead,
+    CgosI2CReadRegister, CgosI2CSetFrequency, CgosI2CType, CgosI2CWrite, CgosI2CWriteReadCombined,
+    CgosI2CWriteRegister, CGOS_I2C_TYPE_DDC, CGOS_I2C_TYPE_PRIMARY, CGOS_I2C_TYPE_SMB,
+    CGOS_I2C_TYPE_UNKNOWN,
+};
+
+/// Error type for I2c operations
+#[derive(PartialEq, Eq, Clone, Debug)]
+pub enum I2cErr {
+    /// User-supplied index is out of range
+    IdxOutOfRange,
+
+    /// I2c bus transaction failed
+    BusErr,
+}
+
+pub type I2cResult<T> = Result<T, I2cErr>;
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum I2cType {
+    Unknown,
+    Primary,
+    Smb,
+    Ddc,
+    CongatecInternalUse(u32),
+}
+
+impl Into<u32> for I2cType {
+    fn into(self) -> u32 {
+        match self {
+            Self::Unknown => CGOS_I2C_TYPE_UNKNOWN,
+            Self::Primary => CGOS_I2C_TYPE_PRIMARY,
+            Self::Smb => CGOS_I2C_TYPE_SMB,
+            Self::Ddc => CGOS_I2C_TYPE_DDC,
+            Self::CongatecInternalUse(x) => x,
+        }
+    }
+}
```

Comment:
an implementation of `From` is preferred since it gives you `Into<_>` for free where the reverse isn't true

```suggestion
impl From<I2cType> for u32 {
    fn from(val: I2cType) -> Self {
        match val {
            I2cType::Unknown => CGOS_I2C_TYPE_UNKNOWN,
            I2cType::Primary => CGOS_I2C_TYPE_PRIMARY,
            I2cType::Smb => CGOS_I2C_TYPE_SMB,
            I2cType::Ddc => CGOS_I2C_TYPE_DDC,
            I2cType::CongatecInternalUse(x) => x,
        }
    }
}
```